[DOCS-1453] Docs for run status change automations (#1329)
wandb/docs: content/guides/core/automations/automation-events.md

- **Adding a new alias to a version of an artifact**: Trigger a specific step of your workflow when a new artifact version in a project or collection has a specific label or alias applied. For example, run a series of downstream processing steps when an artifact has the `test-set-quality-check` alias applied.

### Run events

> An automation be trigered by the following events related to runs.

Reviewer: ```suggestion
An automation can be triggered by a change in a [run's status]({{< relref "/guides/runs/#run-states" >}}) or a [metric]({{< relref "/guides/track/log/#what-data-is-logged-with-specific-wb-api-calls" }}).
```

#### Run status change

> Trigger a workflow when a run changes its status to one or more of **Running**, **Finished**, or **Failed**. **Finished** indicates a run that completed successfully. Optionally, you can further limit the runs that can trigger an automation by filtering by the user that started a run or the run's name.

Reviewer: This makes it sound like a run can have multiple statuses at once. Suggest to something like:

```suggestion
Trigger a workflow when a [run's status]({{< relref "/guides/runs/#run-states" >}}) changes. Optionally, you can further limit the runs that can trigger an automation by filtering by the user that started a run or the run's name.
```

Side note: The table I linked to has "Killed" unsure if that is an option still.

Author: The UI doesn't present "Killed" as an option. @ssisk Is this intentional?

Reviewer 2: Correct, there are many different ways for a run to end, but for purposes of this automation, we grouped them together into failed or finished.

Because run status is a property of the entire run, you can create a run status automation only from the the **Automations** page, not from a workspace.

#### Run metrics change
{{% alert %}}
Run metrics automations are currently available only in [W&B Multi-tenant Cloud]({{< relref "/guides/hosting/#wb-multi-tenant-cloud" >}}).
{{% /alert %}}

Trigger a workflow based on a logged value for a metric, either a metric in a run's history or a [system metric]({{< relref "/guides/models/app/settings-page/system-metrics.md" >}}) such as `cpu`, which tracks the percentage of CPU utilization. W&B logs system metrics automatically every 15 seconds.

To set up a run metric automation, you configure how to compare the metric's value with the threshold you specify. Your choices depend on the event type and on any filters you specify. You can configure the threshold in the following ways:

> - **Run metrics threshold met**: Trigger a workflow when for a given metric, a single logged value or the average logged values meets the threshold you specify.

Reviewer: Unsure if we need to have "run" for this and "Run metrics change threshold met" (since it's nestled under "Run events"). 

The main take away is that we track metrics in an experiment. If and/or when they change, we trigger an automation. I think that message gets lost when we add "run" since it's a W&B-specific computation step and nothing to do with the actual ML stuff. 

```suggestion
- **Metrics threshold met**: Trigger a workflow when for a given metric, a single logged value or the average logged values meets the threshold you specify.
```

Author: ![Screenshot 2025-06-06 at 2 18 38 PM](https://github.com/user-attachments/assets/b6733874-b9fc-4098-9443-f1e6feb4a952)
I think I need to call it the same as in the UI. WDYT?

    1. The number of logged values to average across (defaults to 5).
    1. How to compare the values with the threshold.

  For example, trigger an automation when `accuracy` exceeds `.6`.
- **Run metrics change threshold met**: Trigger a workflow when the average logged values of a run change by the absolute or relative threshold you specify. The automation uses two "windows" of values to check whether to start:
  - The _current window_ averages the 10 most recent values by default.
  - The _prior window_ averages the 50 most recent logged values prior to the current window.

  The windows are consecutive and do not overlap.

  To set up the automation, you configure:
    1. The current window (defaults to 10).
    1. The prior window (defaults to 50).
    1. Whether to evaluate the values as relative or absolute (defaults to **Relative**).
    1. How to compare the values with the threshold:
          - Increases by at least
          - Decreases by at least
          - Increases or decreases by at least

Optionally, you can further limit the runs that can trigger an automation by filtering by the user that started a run or the run's name.

{{% alert %}}

> From a line plot in a workspace, you can quickly create an automation from a line plot in the workspace that shows the metric. Hover over the panel, then click the bell icon at the top of the panel. 

Reviewer: ```suggestion
From a line plot in a workspace, you can create an automation from a line plot in the workspace that shows the metric. Hover over the panel, then click the bell icon at the top of the panel. 
```

- A [system metric]({{< relref "/guides/models/app/settings-page/system-metrics.md" >}}) such as `cpu`, which tracks the percentage of CPU utilization. W&B logs system metrics automatically every 15 seconds.

The notification can watch the metric for these events:

> - **Run status change**: Trigger a workflow when a run changes its status to one or more of **Running**, **Failed**, or **Finished**.

Reviewer 2: One tricky thing is that these run status change events aren't available when creating the automation in the workspace - it's only available from the automations tab (since it's not related to the specific metric chosen, it's related to the whole run)

- **Run metrics threshold met**: Trigger a workflow when for a given metric, a single logged value or the average logged values meets the threshold you specify.
- **Run metrics change threshold met**: Trigger a workflow when the average logged values of a run change by the absolute or relative threshold you specify.

Optionally, you can further limit the runs that can trigger an automation by filtering by the user that started a run or the run's name.

> 

Reviewer 2: It may be worth clarifying what "finished" means in the docs, since the term is ambiguous. it could mean "one way or another it finished" - that could include success or failure. Or it could mean "the run succeeded"

Finished for run status change events means the run succeeded. 

I would have preferred different terminology to avoid this ambiguity in the first place, we went this direction since it matches the terminology in other places in W&B.

Author: I added a bit more detail, but where I talk about the states, I do say that it can be Running, Finished, or Failed. That reads to me like it can't be both finished and failed.
